Lib/importlib/_bootstrap_external.py

```diff
@@ -1203,7 +1203,7 @@ def __init__(self, path, *loader_details):
             loaders.extend((suffix, loader) for suffix in suffixes)
         self._loaders = loaders
         # Base (directory) path
-        self.path = path or '.'
+        self.path = path.__fspath__() if hasattr(path, '__fspath__') else path or '.'
```

Comment:
This should become a multi-line statement as there's too much going on to keep it a single line (even if you do switch to `_os.fspath()` as that `or` is not obvious anymore).